ufo/ufo-fixed-scheduler.c

```diff
@@ -433,12 +433,15 @@ join_threads (GList *threads, GError **error)
 {
     GList *it;
     GError *tmp_error = NULL;
+    gboolean propagated = FALSE;
 
     g_list_for (threads, it) {
         tmp_error = g_thread_join (it->data);
 
-        if (tmp_error)
+        if (tmp_error && !propagated) {
             g_propagate_error (error, tmp_error);
+            propagated = TRUE;
+        }
```

Comment:
In case an error is already propagated we should free all subsequent errors with `g_error_free` and maybe even spam `stderr` with the message.

Reply:
Do you mean errors from different threads or the `tmp_error`? If the latter then I think calling `g_propagate_error` is enough, at least that's how I found it in the [example](https://developer.gnome.org/glib/stable/glib-Error-Reporting.html). Anyway, now I made it even cleaner by using glib's `g_clear_error` in 6221e0df888d1cdc1c9ef6b96479dbb64dbd91ab

Reply:
I meant the `tmp_error` which may originate from different threads. Let's say `read` has an error and `write` has an error then you will discard the error from the thread that finishes first and only propagate the last.